src/renderers/dom/ReactDOMStringRenderer.js

```diff
@@ -22,7 +22,7 @@ var ReactPartialRenderer = require('ReactPartialRenderer');
  */
 function renderToString(element) {
   invariant(
-    React.isValidElement(element),
+    Array.isArray(element) || React.isValidElement(element),
```

Comment:
> I'm not sure this is sufficient. For example it could be a nested array. Conversely, it could be an array of invalid things (e.g. array of objects), and we'd want to throw in that case.

I think here can use 'React.Children.toArray' method to deal with this situation. 
1. nested array: Will flatten the array.
`[[<Hello />,<Day />],<Cool />] => [<Hello />,<Dan />,<Cool />]`
2. array of objects: Will throw when passed object does not have an iterator.


> I think the validation should be moved to a further stage at the point where we actually process the next frame. Which is the same strategy we do with Fiber. This way we don’t do validation immediately, but we throw if we meet something invalid.

if disableNewFiberFeatures=false then don't do validation immediately.